src/LightingThread.cpp

```diff
@@ -78,6 +88,7 @@ class cReader :
 		m_BlockTypes(a_BlockTypes),
 		m_HeightMap(a_HeightMap)
 	{
+		std::fill_n(m_BlockTypes, cChunkDef::NumBlocks * 9, E_BLOCK_AIR);
```

Comment:
Do I read this right that it first fills the entire buffer and then overwrites parts with the actual chunk data? Wouldn't is be even better to only fill the skipped parts of the buffer, on line 30?

Reply:
I tried it out and it's 2-3x _slower_ if you fill selectively.

Reply:
That's weird, I'd assume it's caused by the cache being primed by this. Might be worth trying the perf out on an ARM chip to see if it has the same characteristics - the caches there, if present, work in different ways.

Reply:
I don't think it's that weird really. Filling the whole thing is a single call to `memset` whereas filling a section at a time is ~100 different strided loops that are all inteleaved.

If I can get cross-compilation working later I'll try it out on ARM but I don't think it's likely to be that different.

Reply:
Oh, right, the chunk data is interleaved with other chunks' data, didn't think of that. Now it makes more sense, true. Still I'd like to see a comparison on an ARM, since those have completely different RAM access characteristics, so many optimizations like this are actually anti-optimizations on ARMs.

Reply:
I got it cross-compiled for raspberry pi 3 and doing it this way is still ~2x faster.

Reply:
Great :)